safekeeper/src/http/routes.rs

```diff
@@ -181,16 +184,54 @@ async fn timeline_create_handler(mut request: Request<Body>) -> Result<Response<
             .commit_lsn
             .segment_lsn(server_info.wal_seg_size as usize)
     });
-    tokio::task::spawn_blocking(move || {
-        GlobalTimelines::create(ttid, server_info, request_data.commit_lsn, local_start_lsn)
-    })
-    .await
-    .map_err(|e| ApiError::InternalServerError(e.into()))?
-    .map_err(ApiError::InternalServerError)?;
+    GlobalTimelines::create(ttid, server_info, request_data.commit_lsn, local_start_lsn)
+        .await
+        .map_err(ApiError::InternalServerError)?;
 
     json_response(StatusCode::OK, ())
 }
 
+// Create fake timeline + insert some valid WAL. Useful to test WAL streaming
+// from safekeeper in isolation, e.g.
+// pg_receivewal -v -d "host=localhost port=5454 options='-c tenant_id=deadbeefdeadbeefdeadbeefdeadbeef timeline_id=deadbeefdeadbeefdeadbeefdeadbeef'" -D ~/tmp/tmp/tmp
+// (hacking pg_receivewal startpos is currently needed though to make pg_receivewal work)
+async fn create_fake_timeline_handler(_request: Request<Body>) -> Result<Response<Body>, ApiError> {
```

Comment:
I don't really like `create_fake_timeline_handler`:
- `pg_version`, `lsn`, `tenant_id/timeline_id`, etc. are fixed
- we already have another API (`START_WAL_PUSH`) to create timelines and write WAL; it seems that we can craft an external tool to connect to any safekeeper and write anything we want

Should we include it in this PR?

Reply:
Yeah, I agree it is really ad hoc. However, it is useful because it allows to test `START_REPLICATION` in isolation, I added it when reworked `send_wal.rs` while `receive_wal.rs` was still broken. It might be useful for other similar purposes. But I don't insist, can remove it.

Reply:
I'd removed it. I'm also thinking about removing json_ctrl from time to time :)

Reply:
ok

> I'm also thinking about removing json_ctrl

+1